src/app/core/audio/player.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subscriber } from 'rxjs';
+
+export interface AudioMetadata {
+  duration: number;
+  format: string;
+  bitrate: number;
+  frequency: number;
+}
+
+export interface PlaybackMetadata {
+  duration: number;
+  progress: number;
+}
+
+@Injectable()
+export class PlayerService {
+
+  private playing: Subscriber<PlaybackMetadata>;
+  private playingPlayer: AV.Player;
+  private playingData: PlaybackMetadata;
+
+  play(fileOrUrl: File | string): Observable<PlaybackMetadata> {
+    this.stop();
+    return Observable.create((sub: Subscriber<PlaybackMetadata>) => {
+      this.playing = sub;
+
+      let player: AV.Player;
+      if (fileOrUrl instanceof File) {
+        player = AV.Player.fromFile(fileOrUrl);
+      } else {
+        player = AV.Player.fromURL(fileOrUrl);
+      }
+
+      let data = <PlaybackMetadata> {progress: 0};
+      player.on('duration', d => {
+        data.duration = Math.round(d / 1000);
+        sub.next(data);
+      });
+      player.on('progress', p => {
+        data.progress = Math.round(p / 1000);
+        sub.next(data);
+      });
+      player.on('end', () => sub.complete());
+      player.on('error', err => sub.error(err));
+      player.play();
+
+      this.playingPlayer = player;
+      this.playingData = data;
+
+      return () => player.stop();
```

Comment:
This actually worked out pretty well - whenever the caller unsubscribes from this playback observable, the player will stop.